src/main/java/com/gard/testbed/activitySystem/State.java

```diff
@@ -16,21 +16,32 @@ public State(String stateId) {
     }
 
     @Override
-    public String getId() {
-        return id;
+    public boolean isCompleted(IBlackboard player) {
+        return transitions.stream().allMatch(e -> e.isComplete(player));
+    }
+
+    @Override
+    public boolean isActive() {
+        return false;
     }
```

Comment:
I'm thinking about adding a list of dependencies to State - list of incoming transitions which this state depends on being completed before becoming active. I think this would deal with both limiting sequential tasks and the AND logic of 'parallel' tasks. 

The list of 'Active states' defines the current state of player progression (the tasks he is currently engaged in / need completing next) and...

 the list of 'Active transitions' (outgoing transitions from an 'Active state' which are not completed (i.e. no suitable property in blackboard for trigger)) defines the actions that can be taken at current time.